modules/eslint-plugin/src/rules/component-store/require-super-ondestroy.ts

```diff
@@ -0,0 +1,51 @@
+import path = require('path');
+import { createRule } from '../../rule-creator';
+import { TSESTree } from '@typescript-eslint/types';
+
+export const messageId = 'requireSuperOnDestroy';
+type MessageIds = typeof messageId;
+type Options = readonly [];
+
+export default createRule<Options, MessageIds>({
+  name: path.parse(__filename).name,
+  meta: {
+    type: 'problem',
+    ngrxModule: 'component-store',
+    docs: {
+      description:
+        'Overriden ngOnDestroy method in component stores require a call to super.ngOnDestroy().',
+    },
+    schema: [],
+    messages: {
+      [messageId]:
+        "Call super.ngOnDestroy() inside a component store's ngOnDestroy method.",
+    },
+  },
+  defaultOptions: [],
+  create: (context) => {
+    const ngOnDestroyMethodSelector = `MethodDefinition[key.name='ngOnDestroy']`;
+    const componentStoreClassName = 'ComponentStore';
+
+    let hasNgrxComponentStoreImport = false;
+
+    return {
+      [`ImportDeclaration[source.value='@ngrx/component-store'] ImportSpecifier[imported.name='${componentStoreClassName}']`](
+        _: TSESTree.ImportSpecifier
+      ) {
+        hasNgrxComponentStoreImport = true;
+      },
+      [`ClassDeclaration[superClass.name=${componentStoreClassName}] ${ngOnDestroyMethodSelector}:not(:has(CallExpression[callee.object.type='Super'][callee.property.name='ngOnDestroy'])) > .key`](
```

Comment:
this uses `:has` - which is technically supported by the underlying `esquery`, but is not officially supported by `eslint`. It's not mentioned in t[he docs here](https://eslint.org/docs/latest/extend/selectors#what-syntax-can-selectors-have), and also see this issue: https://github.com/eslint/eslint/issues/14076#issuecomment-926983076.

do you think this approach is reasonably future-proff?

Reply:
I was not aware of this - doesn't seem like support for it would be dropped, but maybe it's safer to use another approach.

@timdeschryver If I wanted to make a PR with an alternative to using `:has`, what would be the best way - should we create a new issue and I make a bugfix PR referencing it, or some other way?

Reply:
I prefer to keep it as-is.
There are multiple rules using `:has`, and I like it because it's easy to read.
If it does become deprecated/removed in the future we can update it.